### ISSUE TITLE
"dotnet publish" doesn't add file version, product name in properties of executable

### ISSUE THREAD

**Issue by TimonDDD** [dotnet/cli#12872]
## Steps to reproduce
Create F# console app, target netcoreapp3.0
specify Version, FileVersion, AssemblyVersion, etc in the .fsproj file
run `dotnet publish -f netcoreapp3.0 -r win10-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true --self-contained`
to make executable
## Expected  behavior
executable should have file version, product name, etc in properties
## Actual behavior
executable doesn't have file version, product name, etc in properties
##
`dotnet --list-sdks` output
```
2.2.401 [C:\Program Files (x86)\dotnet\sdk]
3.0.100 [C:\Program Files (x86)\dotnet\sdk]
```

An example of .fsproj file
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <AssemblyVersion>1.0.0.99</AssemblyVersion>
    <FileVersion>1.0.0.99</FileVersion>
    <Version>1.0.0.99</Version>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Program.fs" />
  </ItemGroup>
</Project>
```
The same command against C# console app produces executable with file version.

**Comment by livarcocc** [comment]
@KevinRansom @brettfo any idea what might be going on here?

**Comment by brettfo** [comment]
I'll be honest, I'm absolutely baffled as to what's happening.  I created two console apps, `version` for F# and `versioncs` for C#.  Properties for each assembly:
![image](https://user-images.githubusercontent.com/926281/66853740-d6a53580-ef34-11e9-8900-dc8a479df0f8.png)

But looking at the manifests from ILDAsm for `version` (F#):

```
//...
.assembly version
{
  .custom instance void [FSharp.Core]Microsoft.FSharp.Core.FSharpInterfaceDataVersionAttribute::.ctor(int32,
                                                                                                      int32,
                                                                                                      int32) = ( 01 00 02 00 00 00 00 00 00 00 00 00 00 00 00 00 ) 
  .custom instance void [System.Runtime]System.Runtime.Versioning.TargetFrameworkAttribute::.ctor(string) = ( 01 00 18 2E 4E 45 54 43 6F 72 65 41 70 70 2C 56   // ....NETCoreApp,V
                                                                                                              65 72 73 69 6F 6E 3D 76 33 2E 30 01 00 54 0E 14   // ersion=v3.0..T..
                                                                                                              46 72 61 6D 65 77 6F 72 6B 44 69 73 70 6C 61 79   // FrameworkDisplay
                                                                                                              4E 61 6D 65 00 )                                  // Name.
  .custom instance void [System.Runtime]System.Reflection.AssemblyCompanyAttribute::.ctor(string) = ( 01 00 07 76 65 72 73 69 6F 6E 00 00 )             // ...version..
  .custom instance void [System.Runtime]System.Reflection.AssemblyConfigurationAttribute::.ctor(string) = ( 01 00 05 44 65 62 75 67 00 00 )                   // ...Debug..
  .custom instance void [System.Runtime]System.Reflection.AssemblyFileVersionAttribute::.ctor(string) = ( 01 00 08 31 2E 32 2E 33 2E 34 34 00 00 )          // ...1.2.3.44..
  .custom instance void [System.Runtime]System.Reflection.AssemblyInformationalVersionAttribute::.ctor(string) = ( 01 00 09 31 2E 32 2E 33 2E 34 34 34 00 00 )       // ...1.2.3.444..
  .custom instance void [System.Runtime]System.Reflection.AssemblyProductAttribute::.ctor(string) = ( 01 00 07 76 65 72 73 69 6F 6E 00 00 )             // ...version..
  .custom instance void [System.Runtime]System.Reflection.AssemblyTitleAttribute::.ctor(string) = ( 01 00 07 76 65 72 73 69 6F 6E 00 00 )             // ...version..
  .custom instance void [System.Runtime]System.Reflection.AssemblyVersionAttribute::.ctor(string) = ( 01 00 07 31 2E 32 2E 33 2E 34 00 00 )             // ...1.2.3.4..

  // --- The following custom attribute is added automatically, do not uncomment -------
  //  .custom instance void [System.Runtime]System.Diagnostics.DebuggableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggableAttribute/DebuggingModes) = ( 01 00 03 01 00 00 00 00 ) 

  .hash algorithm 0x00008004
  .ver 1:2:3:4
}
//...
```

and `versioncs` (C#):

```
//...
.assembly versioncs
{
  .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilationRelaxationsAttribute::.ctor(int32) = ( 01 00 08 00 00 00 00 00 ) 
  .custom instance void [System.Runtime]System.Runtime.CompilerServices.RuntimeCompatibilityAttribute::.ctor() = ( 01 00 01 00 54 02 16 57 72 61 70 4E 6F 6E 45 78   // ....T..WrapNonEx
                                                                                                                   63 65 70 74 69 6F 6E 54 68 72 6F 77 73 01 )       // ceptionThrows.

  // --- The following custom attribute is added automatically, do not uncomment -------
  //  .custom instance void [System.Runtime]System.Diagnostics.DebuggableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggableAttribute/DebuggingModes) = ( 01 00 07 01 00 00 00 00 ) 

  .custom instance void [System.Runtime]System.Runtime.Versioning.TargetFrameworkAttribute::.ctor(string) = ( 01 00 18 2E 4E 45 54 43 6F 72 65 41 70 70 2C 56   // ....NETCoreApp,V
                                                                                                              65 72 73 69 6F 6E 3D 76 33 2E 30 01 00 54 0E 14   // ersion=v3.0..T..
                                                                                                              46 72 61 6D 65 77 6F 72 6B 44 69 73 70 6C 61 79   // FrameworkDisplay
                                                                                                              4E 61 6D 65 00 )                                  // Name.
  .custom instance void [System.Runtime]System.Reflection.AssemblyCompanyAttribute::.ctor(string) = ( 01 00 09 76 65 72 73 69 6F 6E 63 73 00 00 )       // ...versioncs..
  .custom instance void [System.Runtime]System.Reflection.AssemblyConfigurationAttribute::.ctor(string) = ( 01 00 05 44 65 62 75 67 00 00 )                   // ...Debug..
  .custom instance void [System.Runtime]System.Reflection.AssemblyFileVersionAttribute::.ctor(string) = ( 01 00 08 31 2E 32 2E 33 2E 34 34 00 00 )          // ...1.2.3.44..
  .custom instance void [System.Runtime]System.Reflection.AssemblyInformationalVersionAttribute::.ctor(string) = ( 01 00 09 31 2E 32 2E 33 2E 34 34 34 00 00 )       // ...1.2.3.444..
  .custom instance void [System.Runtime]System.Reflection.AssemblyProductAttribute::.ctor(string) = ( 01 00 09 76 65 72 73 69 6F 6E 63 73 00 00 )       // ...versioncs..
  .custom instance void [System.Runtime]System.Reflection.AssemblyTitleAttribute::.ctor(string) = ( 01 00 09 76 65 72 73 69 6F 6E 63 73 00 00 )       // ...versioncs..
  .hash algorithm 0x00008004
  .ver 1:2:3:4
}
//...
```

Both have assembly-level attributes for:

- `AssemblyCompanyAttribute`
- `AssemblyFileVersionAttribute`
- `AssemblyInformationalVersionAttribute`
- `AssemblyProductAttribute`
- `AssemblyTitleAttribute`

The only difference I can see is that F# has an `AssemblyVersionAttribute`, but even removing that doesn't seem to fix the issue.

@KevinRansom do you have any ideas?

**Comment by dasMulli** [comment]
I think https://github.com/dotnet/fsharp/pull/7024 has not been a part of the F# compiler that is included in the 3.0.100 SDK, but I may be wrong.

**Comment by brettfo** [comment]
@dasMulli You're correct.  The 3.0.100 SDK doesn't contain dotnet/fsharp#7024 which embeds the native resources that explorer.exe uses to populate those values, however, that fix will be in the next release of the 3.1 SDK.  I confirmed this locally by building the F# branch `release/dev16.4` which is what will ship in 3.1.

**Comment by livarcocc** [comment]
Thanks for confirming @brettfo and thanks for pointing it out @dasMulli.
